src/components/recordedit/select-all-row.tsx

```diff
@@ -0,0 +1,405 @@
+     */
+    const setValueForAllInputs = (clearValue?: boolean) => {
+        const cm = columnModels[columnModelIndex];
+
+        activeForms?.forEach((formValue: number) => {
+            // ignore the ones that cannot be updated
+            if (appMode === appModes.EDIT && canUpdateValues && !canUpdateValues[`${formValue}-${cm.column.name}`]) {
+                return;
+            }
+            reset(
+                copyOrClearValue(
+                    cm,
+                    getValues(),
+                    foreignKeyData.current,
+                    formValue,
+                    SELECT_ALL_INPUT_FORM_VALUE,
+                    clearValue
+                )
+            );
+        });
+    };
+
+    /* This is to set the width of text area as the width of select-all-row. We have to involve javascript as 
+    the immidiate parent centre-align we cant set a width to it as 100%. So we have to involve JS to set the width of textarea
+    to the next immediate parent width which is select-all-row */
+    const updateTextareaWidth = () => {
+        const textarea = document.querySelector('.select-some-textarea') as HTMLElement;
+        const nonScrollableDiv = document.querySelector('.select-all-row') as HTMLElement;
+        if (textarea) {
+            if (window.innerWidth < 1800) {
+                const newContainerWidth = nonScrollableDiv.offsetWidth;
+                textarea.style.width = `${newContainerWidth}px`;
+            } else {
+                textarea.style.width = '1200px';
+            }
+        }
+    };
+    // -------------------------- render logic ---------------------- //
+
+    const columnModel = columnModels[columnModelIndex];
+    const colName = columnModel.column.name;
+    const inputName = `${SELECT_ALL_INPUT_FORM_VALUE}-${colName}`;
+
+    const btnClass = `${makeSafeIdAttr(
+        columnModel.column.displayname.value
+    )} chaise-btn chaise-btn-secondary`;
+
+    const renderHelpTooltip = () => {
+        const splitLine1 =
+            'You can click on the form to select it and apply changes to it.';
+        const splitLine2 =
+            'You can use the checkbox to select and deselect all records. ' +
+            'By default, if there is no previous selection, the first form will be selected. ';
+        const splitLine3 =
+            'After the forms are selected, ' +
+            'you can click Apply button to apply the changes to selected records.';
+        const splitLine4 =
+            'You can also clear the values for selected records by clicking on Clear button.';
+        return (
+            <>
+                <p>{splitLine1}</p>
+                <p>{splitLine2}</p>
+                <p>{splitLine3}</p>
+                <p>{splitLine4}</p>
+            </>
+        );
+    };
+
+    return (
+        <div className='select-all-row match-entity-value'>
+            <div className={`centre-align ${columnModel.inputType === 'longtext'
```

Comment:
You've done the similar check for inputType in multiple places. Let's just do this once and store it in a variable and then use it in these different places. You can call it `isTextarea` or something like that.